Maps Screen with GPS location

AlecGhost opened the pull request:
### Overview

This PR introduces a map screen for individual events that shows the event's location and the user's location (using the GPS sensor) if location services are enabled.

### Implementation

- `Map` composable that can be used interactively for the screen and later embedded in the event details.
- `MapScreen` with an interactive map and a back button.
- `LocationPermissionRequest` composable that launches the dialog requesting the location permissions from the user
  - If the location is not granted, the map screen still works but does just not display the location.
- Extracted the back button into its own composable because it is now reused in 3 places.
- Update manifest to allow for location usage and add required dependencies

### Testing

- Tested all paths of Map
- For the MapScreen there is not much to do
- The `LocationPermissionRequest` is hard to test, since it involves dialogs. We can grant the permission upfront, then the code runs just fine, but we cannot deny it, which means we cannot test all cases.

### Notes

I did not implement the navigation to and from the Map screen yet,
because it relies on #167 and might mess with #166 which are still open.

If the user changes the orientation of the map, a compass shows in the top left corner, where the back button is. We should adapt our UI design to avoid this overlap (@KianBahia).

There's quite a lot of formatting changes triggered, you can just ignore those.

Resolves: #158 #159
Part of: #124 

<img width="237" height="519" alt="image" src="https://github.com/user-attachments/assets/70125638-99e9-4393-9ec3-4b53ace4296c" />
Note the blue dot at EPFL station is the user's location.

pukoa420 commented on app/src/main/java/ch/epfllife/ui/composables/BackButton.kt:
```diff
@@ -0,0 +1,32 @@
+import ch.epfllife.R
+
+@Composable
+fun BackButton(modifier: Modifier, onGoBack: () -> Unit) {
```
Nitpick: Add default Modifier (modifier: Modifier = Modifier)

pukoa420 reviewed (approved):
Just a few Nitpicks. Implementation looks solid.
Maybe add a little more documentation for the tests in the future but other then that ready to merge

AlecGhost commented on app/src/androidTest/java/ch/epfllife/ui/composables/MapTest.kt:
```diff
@@ -0,0 +1,83 @@
+  // The logic is actually tested by providing a separate callback
+  // that just returns true or false
+  @get:Rule
+  val permissionRule: GrantPermissionRule? =
```
Good question. I didn't give a type at all at the start, but I got the linting error that `GrantPermissionRule` is a Java class and as such nullable and I should therfore make it clear with a type declaration.

AlecGhost commented:
@pukoa420 Thanks for the review! I addressed your comments (either with a comment or through fixing them). I also had to rebase. Please have a look again if everything is ok.

AlecGhost commented on app/src/main/java/ch/epfllife/ui/composables/SystemRequests.kt:
```diff
@@ -0,0 +1,21 @@
+
+  // Request location permission on first composition,
+  // but dialog is skipped if the user set a value previously
+  LaunchedEffect(Unit) {
```
See #176, where I mention implementing the location request model for the embedded view.
